src/c++/perf_analyzer/genai-perf/genai_perf/plots/parser.py

```diff
@@ -0,0 +1,196 @@
+        self._filename = filename
+
+    def generate_configs(self) -> list[PlotConfig]:
+        """Load YAML configuration file and convert to PlotConfigs."""
+        configs = load_yaml(self._filename)
+
+        plot_configs = []
+        for _, config in configs.items():
+            # Collect profile run data
+            profile_data: list[ProfileRunData] = []
+            for filepath in config["paths"]:
+                stats = self._get_statistics(filepath)
+                profile_data.append(
+                    ProfileRunData(
+                        name=self._get_run_name(Path(filepath)),
+                        x_metric=self._get_metric(stats, config["x_metric"]),
+                        y_metric=self._get_metric(stats, config["y_metric"]),
+                    )
+                )
+
+            plot_configs.append(
+                PlotConfig(
+                    title=config["title"],
+                    data=profile_data,
+                    x_label=config["x_label"],
+                    y_label=config["y_label"],
+                    type=self._get_plot_type(config["type"]),
+                    output=Path(config["output"]),
+                )
+            )
+
+        return plot_configs
+
+    def _get_statistics(self, filepath: Path) -> Statistics:
+        """Extract a single profile run data."""
+        profile_filepath = Path(filepath)
+        data_parser = LLMProfileDataParser(
+            filename=profile_filepath,
+            tokenizer=get_tokenizer(DEFAULT_TOKENIZER),
+        )
+        load_info = data_parser.get_profile_load_info()
+
+        # TMA-1904: Remove single experiment assumption
+        assert len(load_info) == 1
+        infer_mode, load_level = load_info[0]
+        stats = data_parser.get_statistics(infer_mode, load_level)
+        return stats
+
+    def _get_run_name(self, filepath: Path) -> str:
+        """Construct a profile run name."""
+        if filepath.parent.name:
+            return filepath.parent.name + "/" + filepath.stem
+        return filepath.stem
+
+    def _get_metric(self, stats: Statistics, name: str) -> list[int | float]:
+        if not name:  # no metric
+            return []
+        elif name == "inter_token_latencies":
+            # Flatten ITL since they are grouped by request
+            itl_flatten = []
+            for request_itls in stats.metrics.data[name]:
+                itl_flatten += request_itls
+            return [scale(x, (1 / 1e9)) for x in itl_flatten]  # ns to ms
+        elif name == "token_positions":
+            token_positions: list[int | float] = []
+            for request_itls in stats.metrics.data["inter_token_latencies"]:
+                token_positions += list(range(1, len(request_itls) + 1))
+            return token_positions
+        elif name == "time_to_first_tokens":
+            ttfts = stats.metrics.data[name]
+            return [scale(x, (1 / 1e9)) for x in ttfts]  # ns to ms
```

Comment:
what does the # ns to ms comment mean?

Reply:
it's nanosecond to millisecond (but I think the code is wrong, the factor should be `1/1e6`)

Reply:
Fixed the code.

Reply:
nice!